docs/build_docs.py

```diff
@@ -0,0 +1,898 @@
+
+    Returns
+    -------
+    logger : logging.Logger
+        The configured logger instance
+    """
+    if use_rich:
+        install_rich_tracebacks()
+        logging_handlers = [RichHandler(rich_tracebacks=True)]
+    else:
+        logging_handlers = [logging.StreamHandler()]
+
+    logging.basicConfig(level=logging.DEBUG, datefmt="[%X]", handlers=logging_handlers)
+    logger = logging.getLogger("build_docs")
+    return logger
+
+
+def parse_args_with_argparse():
+    """
+    Parse and validate command-line arguments provided to the script.
+
+    Returns
+    -------
+    args : argparse.Namespace
+        An object containing the parsed command-line arguments.
+    """
+    # Initialize parser with an example usage in the description
+    parser = argparse.ArgumentParser(
+        description=(
+            "Build Sphinx documentation for GeoIPS related packages.\n"
+            "Example: build_docs.py /path/to/repo geoips"
+        ),
+        formatter_class=rich_argparse.RichHelpFormatter,
+    )
+
+    # Required positional arguments
+    parser.add_argument(
+        "package_name",
+        type=str,
+        help="Name of the package to build (e.g., geoips, data_fusion).",
+    )
+
+    # Optional argument: path to documentation templates
+    parser.add_argument(
+        "--geoips-docs-path",
+        type=str,
+        default=None,
+        help="Path to GeoIPS documentation templates.",
+    )
+    parser.add_argument(
+        "--license-url",
+        type=str,
+        default=None,
+        help=(
+            "Path to GeoIPS license."
+            "Default is https://github.com/NRLMMD-GEOIPS/[package_name]"
+        ),
+    )
+
+    parser.add_argument(
+        "--output-dir",
+        type=str,
+        default=None,
+        help="Output dir to write built docs to",
+    )
+    parser.add_argument(
+        "--repo-path",
+        type=str,
+        default=None,
+        help="Path to the repository (e.g., /path/to/geoips)."
+        "If not provided uses [package_name] to find the packages directory"
+        "Must be an existing directory.",
+    )
+
+    # Parse arguments
+    args = parser.parse_args()
+
+    if not args.repo_path:
+        try:
+            package_path = os.path.dirname(
+                str(importlib.resources.files(args.package_name))
+            )
+            if not package_path:
+                raise ModuleNotFoundError
+            pygit2.Repository(package_path)
+            args.repo_path = package_path
+        except ModuleNotFoundError as e:
+            raise e(f"Could not automatically find repo_path for {args.package_name}.")
+        except pygit2.GitError as e:
+            raise e(
+                "Could not automatically find usable repo_path for "
+                f"{args.package_name}. Found {package_path} but it is not a git repo"
+            )
+
+    if not args.output_dir:
+        output_dir = os.getenv("GEOIPS_DOCSDIR", None)
```

Comment:
Huh, was this used before?  Also, isn't there a deprecation function or decorator or something that officially/explicitly logs deprecated functionality?

Reply:
https://stackoverflow.com/questions/2536307/decorators-in-the-python-standard-lib-deprecated-specifically apparently there isn't, but we could start using the `deprecation` library or write out own decorators

Reply:
We'll probably want to address all of our deprecations at some point.  Hmm, did you know about this?

https://github.com/NRLMMD-GEOIPS/geoips/blob/0e6348b28c72a2ade2ba6230a2ce030ae789a5ae/geoips/geoips_utils.py#L153

Reply:
No I didn't! Should I use that here? It would require that geoips always be installed to build docs for plugins

Reply:
Nope, let's not do that now, but we can use the DeprecationWarning in the future directly maybe!